src/vma/sock/sockinfo.h

```diff
@@ -333,36 +332,32 @@ class sockinfo : public socket_fd_api, public pkt_rcvr_sink, public pkt_sndr_sou
 	void            process_timestamps(mem_buf_desc_t* p_desc);
 
 	virtual bool try_un_offloading(); // un-offload the socket if possible
-#ifdef DEFINED_SOCKETXTREME	
-	virtual inline void do_wakeup()
-	{
-		/* TODO: Let consider if we really need this check */
-		if (!check_vma_active()) {
+
+	virtual inline void do_wakeup()	{
+		if (!is_socketxtreme()) {
 			wakeup_pipe::do_wakeup();
 		}
 	}
 
-	inline bool check_vma_active(void)
-	{
-		return (m_p_rx_ring && m_p_rx_ring->get_vma_active());
+	inline bool is_socketxtreme() {
+		return (m_p_rx_ring && m_p_rx_ring->is_socketxtreme());
```

Comment:
Should be replaced with const class memeber with value of safe_mce_sys().enable_socketxtreme.

Reply:
this construction has two benefits:
1. it is more strong and do not allow to execute socketxtreme flow in case `m_p_rx_ring` is not valid. Socketxtreme can not work w/o m_p_rx_ring
2. it allows to provide run-time detection that can be used in the future (w/o using environment variable). There was a discussion about this possibility in past.

Reply:
i dont agree. they same can be said on ring.h

1. so we should modify the function name - is_socketxtreme_ring ?